pkg/device/device.go

```diff
@@ -0,0 +1,153 @@
+// limitations under the License.
+
+// Package device contains a shim definition of Device to insulate the onos-config subsystem from
+// the deprecation of onos-topo/api/device.
+package device
+
+import (
+	"github.com/onosproject/onos-api/go/onos/topo"
+	"time"
+)
+
+// ID represents device globally unique ID
+type ID topo.ID
+
+// Type represents device type
+type Type string
+
+// Role represents device role in the network
+type Role string
+
+// Device structure provide a shim for topo.Object
+type Device struct {
+	// globally unique device identifier; maps to Object.ID
+	ID ID
+
+	// host:port of the device
+	Address string
+	// device target
+	Target string
+	// device software version
+	Version string
+
+	// timeout indicates the device request timeout
+	Timeout *time.Duration
+	// credentials for connecting to the device
+	Credentials Credentials
+	// TLS configuration for connecting to the device
+	TLS TLSConfig
+
+	// type of the device
+	Type Type
+	// role for the device
+	Role Role
+
+	Protocols []*topo.ProtocolState
+	// user-friendly tag
+	Displayname string
+
+	// arbitrary mapping of attribute keys/values
+	Attributes map[string]string
+
+	// revision of the underlying Object
+	Revision topo.Revision
+}
+
+// Credentials is the device credentials
+type Credentials struct {
+	// user with which to connect to the device
+	User string
+	// password for connecting to the device
+	Password string
+}
+
+// TLSConfig contains information pertinent to establishing a secure connection
+type TLSConfig struct {
+	// name of the device's CA certificate
+	CaCert string
+	// name of the device's certificate
+	Cert string
+	// name of the device's TLS key
+	Key string
+	// indicates whether to connect to the device over plaintext
+	Plain bool
+	// indicates whether to connect to the device with insecure communication
+	Insecure bool
+}
+
+// ListResponse carries a single device event
+type ListResponse struct {
+	// type of the event
+	Type ListResponseType
+	// device is the device on which the event occurred
+	Device *Device
+}
+
+// ListResponseType is a device event type
+type ListResponseType int32
+
+const (
+	// ListResponseNONE obviously
+	ListResponseNONE ListResponseType = 0
+	// ListResponseADDED obviously
+	ListResponseADDED ListResponseType = 1
+	// ListResponseUPDATED obviously
+	ListResponseUPDATED ListResponseType = 2
+	// ListResponseREMOVED obviously
+	ListResponseREMOVED ListResponseType = 3
+)
+
+// ToObject converts topology object entity to local device
+func ToObject(device *Device) *topo.Object {
+	o := &topo.Object{
+		ID:         topo.ID(device.ID),
+		Revision:   device.Revision,
+		Attributes: device.Attributes,
+		Type:       topo.Object_ENTITY,
+		Obj: &topo.Object_Entity{
+			Entity: &topo.Entity{
+				KindID:    "none",
+				Protocols: device.Protocols,
+			},
+		},
+	}
+
+	if o.Attributes == nil {
+		o.Attributes = make(map[string]string)
+	}
+
+	o.Attributes[topo.Type] = string(device.Type)
```

Comment:
@tomikazi I think part of the problem in the integration tests is the mapping between `Type`, `Version` and `Id` of `Device` which are fundamental to `onos-config` and the corresponding placeholders in the new Topo Entity-Kind model.

In the example topology loaded by `onos-umbrella` at 
https://github.com/onosproject/onos-helm-charts/blob/a764d20b4d7f8725ffb6b5f8609dece1d3ce2d5c/onos-umbrella/files/configs/berlin-grid-4-topo.yaml#L27
you can see that I did not add Type as an attribute, as I presumed it was already defined in the Entity/KindID. 

I see however that in this patch, you expect an explicity defined attribute called "Type", and the 'KindID' is set to none.

I think in general when Topo Entities are being synced to `onos-config`, if they do not have an explicit "Type" and "Version" attribute, then they should be ignored as being "**non-config**". It is always possible that there are non-config Entities in `onos-topo` as Topo will have more uses than just driving `onos-config`

Reply:
I've remapped KindId to Type and vice-versa.